Kitodo-DataManagement/src/main/java/org/kitodo/data/database/beans/Comment.java

```diff
@@ -234,12 +235,22 @@ public Task getCorrectionTask() {
     /**
      * Set correctionTask.
      *
-     * @param correctionTask as org.kitodo.data.database.beans.Task
+     * @param correctionTask
+     *         as org.kitodo.data.database.beans.Task
      */
     public void setCorrectionTask(Task correctionTask) {
         this.correctionTask = correctionTask;
     }
 
+    /**
+     * Check if the comment has a correction task which differs from the current task.
+     *
+     * @return true if the comment has a separate correction task
+     */
+    public boolean hasSeparateCorrectionTask() {
```

Comment:
I do not understand the requirement for this method. I thought the `correctionTask` of a comment is _always_ different from it's `currentTask`, and in addition to that _always_ has a lower `ordering` than the `currentTask`. 

@IkramMaalej is that correct?

That's how adding a correction comment via the metadata editor or task details page works at the moment, anyway: only workflow tasks **prior** to the current task are offered as candiates for the "correction task" of a comment. AFAIK that's the whole reasoning behind correction comments and workflows and I wouldn't want to change this rule and logic just to enable updating process states via ActiveMQ. Instead, a `ProcessorException` should be thrown when trying to create a correction comment with a correction task that equals the current task.

Reply:
I added a new behavior to handle errors without specific correction task.

The behavior with the selection of the correction task in creating comment form remains. For external systems that report an error via the new queue, however, it is up to them whether they specify a correction task identifier or not. If none is set then the current task is used as correction task.

fyi @IkramMaalej

Reply:
The concept behind a  correction comment that an error schould be found in a specific workflow step (current task) in a previous workflow step (correction task) and a correction workflow will be started.

I would suggest to implement the need to report an error from external systems  otherwise. 
When `correctionTaskId` is set, then it is simple. A correction comment will be created and a correction workflow will be started. 
If no `correctionTaskId` is set, a simple comment will be created with an additional label (for example `Error_OPEN_EXTERNAL_SYSTEM` or `Error_CLOSE_EXTERNAL_SYSTEM`...) to the comment's message and the status of the current task will be changed (for example to `locked` in this case :  https://github.com/kitodo/kitodo-production/wiki/Developer_3.x-Active-MQ#error_open) . So, we can keep the same behaviour of the correction workflows also for external systems.

Reply:
> If no correctionTaskId is set, a simple comment will be created with an additional label (for example Error_OPEN_EXTERNAL_SYSTEM or Error_CLOSE_EXTERNAL_SYSTEM...) to the comment's message and the status of the current task will be changed (for example to locked in this case : https://github.com/kitodo/kitodo-production/wiki/Developer_3.x-Active-MQ#error_open) . So, we can keep the same behaviour of the correction workflows also for external systems.

For that i added a new task edit type QUEUE so that you can see that the change of task take place from the queue. In addition, the specification of a message is reporting for the reporting of an error over the new processor. So i think these labels are not transparent or user-friendly. 

To prevent npe and fill the empty field in database i added the new behaviour with `correctionTaskId` same like `currentTaskId`. I can change the behaviour to the result that correction task id is not set. If i get an approve for that on this PR i will change that. I don't think the implementation with the labels makes sense so i so I will not adjust this for now.

Correct me if I have misunderstood something :)